### PR TITLE
fix(Kotlin/JS): replace deprecated Kotlin/JS IR compiler API with a n…

### PR DESCRIPTION
**Pull request by JSMonk**
The PR provides a new way to compile JS with IR.
The currently used API will be removed from the compiler soon, so the provided changes will prevent breaks on the compiler API side.

### CODE REVIEW

**Comment by AlexanderPrendota** [comment]
should I merge it? =)

**Comment by JSMonk** [comment]
Sure. We've just started to delete the described API, so, it will be great if we will merge it before compiler updates that will not contain deprecated API.

**Comment by AlexanderPrendota** [comment]
Okay, do we need to have test for such changes?

File: src/main/kotlin/com/compiler/server/compiler/components/KotlinToJSTranslator.kt [hunk 1]
```diff
@@ -115,19 +117,25 @@ class KotlinToJSTranslator(
...
+      dce = false,
+      perModule = false,
+      minimizedMemberNames = false
     )
 
-    val compiledModule: CompilerResult = transformer.generateModule(ir.allModules)
-    val jsCode = compiledModule.outputs.values.single().jsCode
+    val compiledModule: CompilerResult = transformer.generateModule(ir.allModules, setOf(translationMode))
+
+    val jsCode = compiledModule.outputs[translationMode]!!.jsCode
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/KotlinToJSTranslator.kt:138]: is it possible to avoid using `!!`?

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/compiler/server/compiler/components/KotlinToJSTranslator.kt (hunk 1 above)
Comment (AlexanderPrendota) [reply src/main/kotlin/com/compiler/server/compiler/components/KotlinToJSTranslator.kt:138]: cc @JSMonk

**Comment by AlexanderPrendota** [comment]
@JSMonk ping

File: src/main/kotlin/com/compiler/server/compiler/components/KotlinToJSTranslator.kt (hunk 1 above)
Comment (AlexanderPrendota) [reply src/main/kotlin/com/compiler/server/compiler/components/KotlinToJSTranslator.kt:138]: yes, let's do it

**Comment by AlexanderPrendota** [comment]
@JSMonk friendly ping

**Pull request closed without merging**